route-rs-runtime/src/element/utils/icmp.rs

```diff
@@ -0,0 +1,262 @@
+use smoltcp::phy::ChecksumCapabilities;
+use smoltcp::wire::*;
+use std::cmp;
+
+/// The size of an ICMP "header", which precedes the data field
+const ICMP_HDR_LEN: usize = 8;
+
+/// Helper to generate ICMPv4 error messages
+pub struct Icmpv4ErrorGenerator {
+    src_ip: Ipv4Address,
+    mtu: usize,
+    ttl: u8,
+    bad_source_addrs: Vec<Ipv4Address>,
+}
+
+impl Icmpv4ErrorGenerator {
+    /// Create an ICMPv4 error generator
```

Comment:
I like the documentation to start on this. Good idea since this is user facing eventually.